sources/packages/backend/apps/api/src/route-controllers/application-exception/_tests_/e2e/application-exception.aest.getExceptionById.e2e-spec.ts

```diff
@@ -0,0 +1,103 @@
+import { HttpStatus, INestApplication } from "@nestjs/common";
+import * as request from "supertest";
+import { DataSource, Repository } from "typeorm";
+import {
+  AESTGroups,
+  BEARER_AUTH_TYPE,
+  createTestingAppModule,
+  getAESTToken,
+  getAESTUser,
+} from "../../../../testHelpers";
+import {
+  createFakeApplicationException,
+  createFakeApplicationExceptionRequest,
+} from "@sims/test-utils";
+import {
+  ApplicationException,
+  ApplicationExceptionRequest,
+  ApplicationExceptionStatus,
+} from "@sims/sims-db";
+import { SystemUsersService } from "@sims/services";
+import { getUserFullName } from "../../../../utilities";
+
+describe(`ApplicationExceptionAESTController(e2e)-getExceptionById`, () => {
+  let app: INestApplication;
+  let applicationExceptionRepo: Repository<ApplicationException>;
+  let applicationExceptionRequestRepo: Repository<ApplicationExceptionRequest>;
+  let systemUsersService: SystemUsersService;
+  let appDataSource: DataSource;
+
+  beforeAll(async () => {
+    const { nestApplication, dataSource, module } =
+      await createTestingAppModule();
+    app = nestApplication;
+    applicationExceptionRepo = dataSource.getRepository(ApplicationException);
+    applicationExceptionRequestRepo = dataSource.getRepository(
+      ApplicationExceptionRequest,
+    );
+    systemUsersService = await module.get(SystemUsersService);
+    appDataSource = dataSource;
+  });
+
+  it("Should get an application exception by id when available.", async () => {
+    // Arrange
+    const creator = await systemUsersService.systemUser();
+    const assessedBy = await getAESTUser(
+      appDataSource,
+      AESTGroups.BusinessAdministrators,
+    );
+
+    let applicationException = createFakeApplicationException(
```

Comment:
Line 50-63 could be a factory/helper as we would use it many times.

Reply:
I refactored that code and I'm using the same helper functions for as in the other tests.